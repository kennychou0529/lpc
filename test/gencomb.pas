program vectorcaract;
label 1;
type vec=array[1..50] of integer;
var i,ig,n,k:integer;
    a,v:vec;

procedure gen(var v:vec);
label 10;
var i,j:0..255;
begin
if ig=0 then begin
             for i:=1 to n do v[i]:=i;
             ig:=1;goto 10;
             end;
for i:=k downto 1 do
      if v[i]<n+i-k then begin v[i]:=v[i]+1;
                           for j:=i+1 to k do  v[j]:=v[j-1]+1;
                              goto 10;
                        end;
ig:=0;
10:end;

procedure prel(v:vec);
var i : integer;
begin
 for i:=1 to k do if v[i]<>0 then write(v[i],'  ');
 writeln;
end;

begin
 writeln('generarea combinarilor de n luate cåte k');
 write('n=');readln(n);writeln(n);
 write('k=');readln(k);writeln(k);
  ig:=0;
 repeat
  gen(v);
  if ig=0 then goto 1;
  prel(v);
 until ig=0;
1:end.
