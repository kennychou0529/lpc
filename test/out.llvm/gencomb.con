
compiling pass 1 (LPC)
-----------------------------------------------------
  warning on line 7: 'v' hiding existing declaration on line 5
  warning on line 9: 'i' hiding existing declaration on line 4
  warning on line 23: 'v' hiding existing declaration on line 5
  warning on line 24: 'i' hiding existing declaration on line 4

0 error(s), 4 warning(s)

P1 errcode = 0

compiling pass 2 (target = llvm)
-----------------------------------------------------

P2 errcode = 0
