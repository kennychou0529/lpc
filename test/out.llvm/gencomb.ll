; ModuleID = 'gencomb.pas'
source_filename = "C:\Users\lemo\work\compilers\lpc\test\gencomb.pas"
target datalayout = "e-m:w-i64:64-f80:128-n8:16:32:64-S128"

; program arguments (command line mapping)
%struct._Filename = type { i8*, i8* }
@_FilenameMapEntries = internal global [3 x %struct._Filename]
    [
    %struct._Filename { i8* getelementptr inbounds ([7 x i8], [7 x i8]* @.str.1, i32 0, i32 0), i8* null },
    %struct._Filename { i8* getelementptr inbounds ([8 x i8], [8 x i8]* @.str.2, i32 0, i32 0), i8* null },
    %struct._Filename { i8* null, i8* null }
    ], align 16
@_FilenameMap = dso_local global %struct._Filename* getelementptr inbounds ([3 x %struct._Filename], [3 x %struct._Filename]* @_FilenameMapEntries, i32 0, i32 0), align 8
@_FilenameMapSize = dso_local constant i32 2, align 4

; line 0
;================================================================================
; scope: global scope (level : 0)

; types
%T_text = type i8*

; line 1
;================================================================================
; scope: program scope (level : 1)

; types
%T_vec = type [50 x i32]

; program variables
@_input = dso_local global %T_text zeroinitializer
@_output = dso_local global %T_text zeroinitializer
@a = dso_local global %T_vec zeroinitializer
@i = dso_local global i32 zeroinitializer
@ig = dso_local global i32 zeroinitializer
@k = dso_local global i32 zeroinitializer
@n = dso_local global i32 zeroinitializer
@v = dso_local global %T_vec zeroinitializer

; procedure body
define void @P_()
{
    ret void
}

; line 7
;================================================================================
; scope: gen (level : 2)

; activation record
%Frame_gen = type
{
    ; parameters
    %T_vec,    ; 0: v

    ; variables
    i32,    ; 1: i
    i32,    ; 2: j

    ; dummy
    i8*
};

; procedure body
define void @P_gen()
{
    %1 = alloca %Frame_gen, align 8
    ret void
}

; line 23
;================================================================================
; scope: prel (level : 2)

; activation record
%Frame_prel = type
{
    ; parameters
    %T_vec,    ; 0: v

    ; variables
    i32,    ; 1: i

    ; dummy
    i8*
};

; procedure body
define void @P_prel()
{
    %1 = alloca %Frame_prel, align 8
    ret void
}

;================================================================================
; string literals

@.str.1 = private unnamed_addr constant [7 x i8] c"_input\00", align 1
@.str.2 = private unnamed_addr constant [8 x i8] c"_output\00", align 1


;================================================================================
; metadata

!llvm.dbg.cu = !{!16}
!llvm.module.flags = !{!17, !18}
!llvm.ident = !{!15}

!0 = !DIFile(filename: "gencomb.pas", directory: "C:\Users\lemo\work\compilers\lpc\test")
!1 = !DIBasicType(name: "boolean", size: 8, encoding: DW_ATE_boolean)
!2 = !DIBasicType(name: "char", size: 8, encoding: DW_ATE_unsigned_char)
!3 = !DIBasicType(name: "integer", size: 32, encoding: DW_ATE_signed)
!4 = !DIDerivedType(tag: DW_TAG_pointer_type, file: !0, line: 0, baseType: !5, size: 64)
!5 = !DIBasicType(name: "void", size: 0)
!6 = !DIBasicType(name: "real", size: 64, encoding: DW_ATE_float)
!7 = !DIDerivedType(tag: DW_TAG_pointer_type, name: "text", file: !0, line: 0, baseType: null, size: 64)
!8 = !DICompositeType(tag: DW_TAG_array_type, name: "vec", file: !0, line: 3, baseType: !3, size: 1600, elements: !10)
!9 = !DISubrange(count: 50)
!10 = !{!9}
!11 = !DIBasicType(name: "range", size: 32, encoding: DW_ATE_signed)
!12 = !{!1,!2,!3,!4,!5,!6,!7,!8,!11}
!13 = !{}
!14 = !{}
!15 = !{!"LPC 1.1"}
!16 = distinct !DICompileUnit(language: DW_LANG_Pascal83, file: !0, producer: "LPC 1.1", isOptimized: true, runtimeVersion: 0, emissionKind: FullDebug, enums: !13, retainedTypes: !12, globals: !14, nameTableKind: None)
!17 = !{i32 2, !"CodeView", i32 1}
!18 = !{i32 2, !"Debug Info Version", i32 3}

