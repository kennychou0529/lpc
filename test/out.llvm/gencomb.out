generarea combinarilor de n luate cï¿½te k
n=6
k=3
67305985  262914  1027  
